titus-common/src/main/java/com/netflix/titus/common/util/guice/internal/ActivationProvisionListener.java

```diff
@@ -44,7 +46,7 @@
 
     private static final Logger logger = LoggerFactory.getLogger(ActivationProvisionListener.class);
 
-    private final List<ServiceHolder> services = new CopyOnWriteArrayList<>();
+    private final Set<ServiceHolder> services = new CopyOnWriteArraySet<>();
```

Comment:
We cannot use set here, as the activation order is important. Deactivation is executed in the reverse order.

Reply:
good point, will fix

Reply:
On a second thought, some `Sets` preserve insertion order, it is implementation specific. The one we are using here does preserve the insertion order, so we are fine.

Unfortunately in Java none of the `Collection` interfaces explicitly capture the intent of preserving insertion order. I will add a comment to make it more explicit.